In [1]:
import numpy as np
import pandas as pd

### Задача 1

#### Даны значения величины заработной платы заемщиков банка (salary) и значения их поведенческого кредитного скоринга (scoring):
#### salary = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110]
#### scoring = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832]

#### 1. Найдите ковариацию этих двух величин: сначала без использования специальных функций,
####    а затем с помощью функции numpy.cov. Полученные значения должны быть равны.
#### 2. Найдите коэффициент корреляции Пирсона: с помощью ковариации и средних квадратических отклонений двух признаков,
####    а затем с помощью специальных функций из библиотек numpy и pandas.

### Решение:

#### Исходные данные:

In [2]:
df = pd.DataFrame({'salary': [35, 45, 190, 200, 40, 70, 54, 150, 120, 110],
                   'scoring': [401, 574, 874, 919, 459, 739, 653, 902, 746, 832]})
df

salary  scoring
0      35      401
1      45      574
2     190      874
3     200      919
4      40      459
5      70      739
6      54      653
7     150      902
8     120      746
9     110      832

#### 1. Найдем ковариацию двух величин:

In [3]:
X = df['salary']
Y = df['scoring']

MX = X.mean()
MY = Y.mean()

cov = ((X - MX) * (Y - MY)).mean()
cov

9157.84

In [4]:
cov = (X * Y).mean() - MX * MY
cov

9157.839999999997

In [5]:
cov = np.cov(X, Y)
cov

array([[ 3882.93333333, 10175.37777778],
       [10175.37777778, 33854.32222222]])

In [6]:
cov = ((X - MX) * (Y - MY)).sum() / (X.shape[0] - 1)
cov

10175.37777777778

#### 2. Найдем коэффициент корреляции Пирсона:

In [7]:
corr = cov / (X.std() * Y.std())
corr

0.8874900920739164

In [8]:
df[['salary', 'scoring']].corr()

salary  scoring
salary   1.00000  0.88749
scoring  0.88749  1.00000

### Задача 2

#### Измерены значения IQ выборки студентов, обучающихся в местных технических вузах:
#### 131, 125, 115, 122, 131, 115, 107, 99, 125, 111

#### Известно, что в генеральной совокупности IQ распределен нормально.
#### Найдите доверительный интервал для математического ожидания с надежностью 0.95.

### Решение:

In [9]:
samples = np.array([131, 125, 115, 122, 131, 115, 107, 99, 125, 111])
samples

array([131, 125, 115, 122, 131, 115, 107,  99, 125, 111])

In [10]:
n = 10

#### Найдем среднее и несмещенную оценку для среднего квадратического отклонения:

In [11]:
m = samples.mean()
m

118.1

In [12]:
s = samples.std(ddof=1)
s

10.54566788359614

#### Нужный нам квантиль найдем по таблице критических значений t-критерия Стьюдента:

In [13]:
t = 2.2622

#### Найдем искомый доверительный интервал:

In [14]:
(m - t * s / np.sqrt(n), m - t * s / np.sqrt(n) )

(110.55594079648245, 110.55594079648245)

### Ответ:

#### Доверительный интервал $110.55594079648245 < a < 110.55594079648245$

### Задача 3

#### Известно, что рост футболистов в сборной распределен нормально с известной дисперсией 25.
#### На выборке объёма 27 выборочное среднее составило 174.2.
#### Найдите доверительный интервал для математического ожидания с надежностью 0.95.

### Решение:

#### По условию задачи известно:

#### среднее квадратическое отклонение $\sigma = 25$, выборочная средняя $х^{\ast } = 174.2$ и объем выборки $n = 27$.

#### Для нахождения доверительного интервала все величины кроме $t$ (аргумента функции Лапласа) известны.

#### Найдем $t$ из соотношения $\Phi(t) = 0,95/2= 0,475$.

#### По таблице значений функции Лапласа находим $t = 1,96$ и
#### $t{\displaystyle \sigma \over\displaystyle \sqrt n } = \delta $ - точность оценки

In [15]:
1.96 * 25 / np.sqrt(27)

9.430054396763888

#### Получаем доверительный интервал:

In [16]:
(174.2 - 9.430054396763888,  174.2 + 9.430054396763888)

(164.7699456032361, 183.63005439676388)

### Ответ:

#### Доверительный интервал $164.7699456032361 < a < 183.63005439676388$